doc/whats_new/v1.0.rst

```diff
@@ -415,6 +415,11 @@ Changelog
 - |FIX| :class:`neighbors.DistanceMetric` subclasses now support readonly
   memory-mapped datasets. :pr:`19883` by `Julien Jerphanion <jjerphan>`.
 
+- |FIX| :class:`neighbors.NearestNeighbors`, :class:`neighbors.KNeighborsClassifier`,
+  :class:`neighbors.RadiusNeighborsClassifier`, :class:`neighbors.KNeighborsRegressor`
+  and :class:`neighbors.RadiusNeighborsRegressor` classes now follow design principles
+  for estimators. :pr:`20072` by :user:`Juan Carlos Alfaro Jiménez <alfaro96>`.
```

Comment:
To be explicit about the change:

```suggestion
  and :class:`neighbors.RadiusNeighborsRegressor` does not validate `weights` in
  `__init__` and validates `weights` in `fit` instead. :pr:`20072` by
  :user:`Juan Carlos Alfaro Jiménez <alfaro96>`.
```

Reply:
Thank you @thomasjpfan for the suggestion 😉.